server/api/impl/configurator/configurator.go

```diff
@@ -86,19 +85,7 @@ func (c *config) RefreshConfig(stored *api.StoredConfig) error {
 	if prevStored != nil {
 		if prevStored.AWSSecretAccessKey != stored.AWSSecretAccessKey ||
 			prevStored.AWSAccessKeyID != stored.AWSAccessKeyID {
-			var creds *credentials.Credentials
-			if stored.AWSSecretAccessKey == "" && stored.AWSAccessKeyID == "" {
-				creds = credentials.NewEnvCredentials() // Read Mattermost cloud credentials from the environment variables
-			} else {
-				creds = credentials.NewStaticCredentials(stored.AWSAccessKeyID, stored.AWSSecretAccessKey, "")
-			}
-
-			if newConfig.AWSSession, err = session.NewSession(&sdk.Config{
-				Region:      sdk.String(aws.DefaultRegion),
-				Credentials: creds,
-			}); err != nil {
-				return err
-			}
+			c.awsClient.RefreshService(stored.AWSAccessKeyID, stored.AWSSecretAccessKey)
```

Comment:
0/5 style see the previous comment, should probably be a separate RefreshAWS, called from `p.OnConfigurationChange` directly